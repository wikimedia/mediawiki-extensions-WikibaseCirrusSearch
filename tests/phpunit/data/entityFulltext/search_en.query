{
	"search": "Duck",
	"userLang": "en",
	"ns": [0]
}
