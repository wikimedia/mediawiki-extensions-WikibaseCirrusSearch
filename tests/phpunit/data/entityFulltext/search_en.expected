{
    "query": {
        "bool": {
            "disable_coord": true,
            "should": [
                {
                    "term": {
                        "title.keyword": "DUCK"
                    }
                },
                {
                    "bool": {
                        "filter": [
                            {
                                "bool": {
                                    "should": [
                                        {
                                            "match": {
                                                "all": {
                                                    "query": "Duck",
                                                    "operator": "AND"
                                                }
                                            }
                                        },
                                        {
                                            "match": {
                                                "all.plain": {
                                                    "query": "Duck",
                                                    "operator": "AND"
                                                }
                                            }
                                        },
                                        {
                                            "match": {
                                                "descriptions.en": {
                                                    "query": "Duck",
                                                    "operator": "AND"
                                                }
                                            }
                                        }
                                    ]
                                }
                            }
                        ],
                        "should": [
                            {
                                "dis_max": {
                                    "tie_breaker": 0,
                                    "queries": [
                                        {
                                            "constant_score": {
                                                "filter": {
                                                    "match": {
                                                        "labels.en.near_match": "Duck"
                                                    }
                                                },
                                                "boost": 0.78
                                            }
                                        },
                                        {
                                            "constant_score": {
                                                "filter": {
                                                    "match": {
                                                        "labels.en.near_match_folded": "Duck"
                                                    }
                                                },
                                                "boost": 0.01
                                            }
                                        }
                                    ]
                                }
                            },
                            {
                                "bool": {
                                    "should": [
                                        {
                                            "match": {
                                                "all": {
                                                    "query": "Duck",
                                                    "operator": "OR",
                                                    "boost": 0.04
                                                }
                                            }
                                        },
                                        {
                                            "match": {
                                                "all.plain": {
                                                    "query": "Duck",
                                                    "operator": "OR",
                                                    "boost": 0.04
                                                }
                                            }
                                        },
                                        {
                                            "dis_max": {
                                                "tie_breaker": 0.2,
                                                "queries": [
                                                    {
                                                        "match": {
                                                            "descriptions.en": {
                                                                "query": "Duck",
                                                                "boost": 0.07
                                                            }
                                                        }
                                                    },
                                                    {
                                                        "match": {
                                                            "labels.en.plain": {
                                                                "query": "Duck",
                                                                "boost": 0.07
                                                            }
                                                        }
                                                    },
                                                    {
                                                        "match": {
                                                            "descriptions.en.plain": {
                                                                "query": "Duck",
                                                                "boost": 0.07
                                                            }
                                                        }
                                                    }
                                                ]
                                            }
                                        }
                                    ]
                                }
                            }
                        ]
                    }
                }
            ],
            "minimum_should_match": 1,
            "filter": [
                {
                    "terms": {
                        "namespace": [
                            0
                        ]
                    }
                }
            ]
        }
    },
    "rescore_query": [
        {
            "window_size": 8192,
            "query": {
                "query_weight": 1,
                "rescore_query_weight": 1,
                "score_mode": "total",
                "rescore_query": {
                    "function_score": {
                        "score_mode": "sum",
                        "functions": [
                            {
                                "script_score": {
                                    "script": {
                                        "source": "pow(doc['incoming_links'].value , 1) \/ ( pow(doc['incoming_links'].value, 1) + pow(100,1))",
                                        "lang": "expression"
                                    }
                                },
                                "weight": 0.6
                            },
                            {
                                "script_score": {
                                    "script": {
                                        "source": "pow(doc['sitelink_count'].value , 2) \/ ( pow(doc['sitelink_count'].value, 2) + pow(20,2))",
                                        "lang": "expression"
                                    }
                                },
                                "weight": 0.4
                            },
                            {
                                "weight": -1,
                                "filter": {
                                    "term": {
                                        "statement_keywords": "P31=Q4167410"
                                    }
                                }
                            }
                        ]
                    }
                }
            }
        }
    ]
}